File: sdks/python/examples/market-data/recent_trades.py
import pmxt

def main():
    # Kalshi
    kalshi = pmxt.Kalshi()
    k_markets = kalshi.fetch_markets(query="Fed Chair")
    k_market = kalshi.filter_markets(k_markets, "Kevin Warsh")[0]

    k_trades = kalshi.fetch_trades(k_market.yes.outcome_id, limit=10)
    print('Kalshi:', k_trades)

    # Polymarket
    poly = pmxt.Polymarket()
    p_markets = poly.fetch_markets(query='Fed Chair')
    p_market = poly.filter_markets(p_markets, 'Kevin Warsh')[0]

    p_trades = poly.fetch_trades(p_market.yes.outcome_id, limit=10)
    print('Polymarket:', p_trades)

if __name__ == "__main__":
    main()

File: sdks/typescript/examples/market-data/recent_trades.ts
import pmxt from 'pmxtjs';

const main = async () => {
    // Kalshi
    const kalshi = new pmxt.Kalshi();
    const kEvent = (await kalshi.searchEvents('Fed Chair'))[0];
    const kMarket = kEvent.searchMarkets('Kevin Warsh')[0];

    const kTrades = await kalshi.fetchTrades(kMarket.yes!.id, { limit: 10, resolution: '1m' });
    console.log('Kalshi:', kTrades);

    // Polymarket
    const poly = new pmxt.Polymarket();
    const pEvent = (await poly.searchEvents('Fed Chair'))[0];
    const pMarket = pEvent.searchMarkets('Kevin Warsh')[0];

    // Use .yes.id for convenience
    const pTrades = await poly.fetchTrades(pMarket.yes!.id, { limit: 10, resolution: '1m' });
    console.log('Polymarket:', pTrades);
};

main();